### PR TITLE
feat: add publish_jobs config field

### PR DESCRIPTION
**Pull request by mistydemeo**
This is a quick spike of the new `publish_jobs` config field. Now, instead of just using `tap` to decide whether to enable a Homebrew publish job, we also have this new field. Right now, we don't explicitly ask the user to enable it and instead implicitly enable it if `tap` is enabled, but we can update the `init` UI for that if we want.

### CODE REVIEW

File: cargo-dist/src/config.rs [hunk 1]
```diff
@@ -221,6 +221,11 @@ pub struct DistMetadata {
     #[serde(skip_serializing_if = "Option::is_none")]
     #[serde(rename = "all-features")]
     pub all_features: Option<bool>,
+
+    /// Publish jobs to run in CI
+    ///
+    /// (defaults to none)
+    pub publish_jobs: Option<Vec<PublishStyle>>,
```
Comment (Gankra) [inline cargo-dist/src/config.rs:228]: `#[serde(rename)]` to kebab case for consistency

File: cargo-dist/src/config.rs [hunk 2]
```diff
@@ -346,6 +353,9 @@ impl DistMetadata {
         if tap.is_none() {
             *tap = workspace_config.tap.clone();
         }
+        if publish_jobs.is_none() {
+            *publish_jobs = workspace_config.publish_jobs.clone();
+        }
```
Comment (Gankra) [inline cargo-dist/src/config.rs:358]: should be a workspace-only field, i think...?

hmm. actually I guess we could make the CI intelligently only publish certain apps to certain venues? if like one only makes sense on pypi or w/e. So I guess package-granularity makes sense.

File: cargo-dist/src/init.rs [hunk 3]
```diff
@@ -475,6 +479,8 @@ fn get_new_dist_metadata(
                 meta.tap = None;
             } else {
                 meta.tap = Some(tap.to_owned());
+                meta.publish_jobs = Some(vec![PublishStyle::Homebrew]);
```
Comment (Gankra) [inline cargo-dist/src/init.rs:482]: We should write this more compositionally. Start with the existing inputs read from the config, then push this value if it's not already in the array (will make things a lot better when we start adding more).

File: cargo-dist/src/init.rs [hunk 4]
```diff
@@ -759,6 +766,13 @@ fn update_toml_metadata(
         *all_features,
     );
 
+    apply_string_list(
+        table,
+        "publish_jobs",
```
Comment (Gankra) [inline cargo-dist/src/init.rs:771]: This will need to be renamed too

File: cargo-dist/templates/ci/github_ci.yml.j2 [hunk 5]
```diff
@@ -170,7 +170,7 @@ jobs:
           gh release upload ${{ github.ref_name }} $(cat uploads.txt)
           echo "uploaded!"
 
-{{%- if tap %}}
+{{%- if 'homebrew' in publish_jobs and tap %}}
```
Comment (Gankra) [inline cargo-dist/templates/ci/github_ci.yml.j2:173]: Somewhere in the rust code should probably error out if the first part is true but the second part isn't. The "npm has to have tar.gz" checks might be a good example to go off of..?

**Review by Gankra — CHANGES_REQUESTED** [review]

File: cargo-dist/src/init.rs (hunk 3 above)
Comment (Gankra) [reply cargo-dist/src/init.rs:482]: in retrospect this one is ignorable if we just want to get it out the door

File: cargo-dist/templates/ci/github_ci.yml.j2 (hunk 5 above)
Comment (Gankra) [reply cargo-dist/templates/ci/github_ci.yml.j2:173]: also ignorable if we just want it out the door

File: cargo-dist/src/config.rs (hunk 2 above)
Comment (mistydemeo) [reply cargo-dist/src/config.rs:358]: Yeah, it seems like there's room for some extra detail there.

**Review by Gankra — APPROVED** [review]

File: cargo-dist/src/init.rs (hunk 3 above)
Comment (mistydemeo) [reply cargo-dist/src/init.rs]: Done.

File: cargo-dist/templates/ci/github_ci.yml.j2 (hunk 5 above)
Comment (mistydemeo) [reply cargo-dist/templates/ci/github_ci.yml.j2:173]: We don't have an error case for this in npm right now except in `init`, and the Homebrew `init` prevents the case from existing. I did add a warning for if it comes up during rendering though, same as with npm.

**Pull request merged**